MailManager/src/tests/java/inc/softserve/EnvelopeServiceTest.java

```diff
@@ -0,0 +1,78 @@
+package inc.softserve;
+
+import inc.softserve.common.EnvelopeTools;
+import inc.softserve.dao.EnvelopeDao;
+import inc.softserve.dao.UserDao;
+import inc.softserve.services.EnvelopeService;
+import inc.softserve.services.EnvelopeServiceImpl;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class EnvelopeServiceTest extends Assert {
+
+    EnvelopeService service;
+
+    @BeforeClass
+    public void setUpDataSource() throws Exception {
+
+        UserDao usrDao = mock(UserDao.class);
+        User user = new User("akonon", "a", "b");
+        when(usrDao.getUserByUsername(anyString())).thenReturn(null);
+        when(usrDao.getUserByUsername("akonon")).thenReturn(user);
+
+        Envelope envelopeNotFound = new Envelope(user, "akonon22@mail.ru", "TestEmail", "Some template", EnvelopeState.NOT_FOUND);
+        Envelope envelopeSent = new Envelope(user, "akonon22@mail.ru", "TestEmail", "Some template", EnvelopeState.SENT);
+        EnvelopeDao envDao = mock(EnvelopeDao.class);
+        when(envDao.saveOrUpdate(any(Envelope.class))).thenAnswer((invocation) -> {Object[] args = invocation.getArguments();
+                                                                                    return (Envelope) args[0];});
+        when(envDao.getById("goodId")).thenReturn(envelopeSent);
+        when(envDao.getById("badId")).thenReturn(envelopeNotFound);
+        when(envDao.getById("illegalId")).thenReturn(null);
```

Comment:
Those scenario-specific set ups should be adjacent to test code that relies on it. I'd suggest to split method `testGetEnvelopeStateById` in three different scenarios and precede related test code with such mockito instructions.